Using APIs w2 sally samaan

Sally-Samaan opened the pull request.

nori0aw commented on 3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js:
```diff
@@ -22,18 +22,88 @@ Use async/await and try/catch to handle promises.
+
+    //gettingData
+
+    function gettingPokemonData() {
+      const selectPokemon = document.getElementById('pokemon-list');
+      selectPokemon.addEventListener('input', getValue);
+      function getValue() {
+        const pokemonIndex = this.value;
+        fetchImage(pokemonURL[pokemonIndex], image);
+      }
+    }
+    gettingPokemonData();
```
Why did you create this function gettingPokemonData() here ? I do not see any use in doing it as a function.!

✏️need to modify✏️

nori0aw commented on 3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js:
```diff
@@ -22,18 +22,88 @@ Use async/await and try/catch to handle promises.
 
-function fetchImage(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function fetchImage(pokemonURL, image) {
+  fetchData(pokemonURL).then((pokemon) => {
+    const pokemonSprite = pokemon.sprites.back_default;
+    image.src = pokemonSprite;
+  });
```
You need to use async await (as per the guidelines) when you are dealing with promises for this exercise.

✏️need to modify✏️

nori0aw commented on 3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js:
```diff
@@ -22,18 +22,88 @@ Use async/await and try/catch to handle promises.
 
 function main() {
-  // TODO complete this function
+  fetchData('https://pokeapi.co/api/v2/pokemon?limit=151')
+    .then((x) => {
+      const pokemonArray = x.results;
+      const pokemonName = pokemonArray.map((x) => {
+        return x.name;
+      });
+      const pokemonURL = pokemonArray.map((x) => {
+        return x.url;
+      });
+
+      fetchAndPopulatePokemons(pokemonName, pokemonURL);
+    })
+    .catch((e) => {
+      console.log(e);
```
Also here you need to use async await (as per the guidelines) when you are dealing with promises for this exercise.

✏️need to modify✏️

nori0aw commented on 3-UsingAPIs/Week2/homework/ex3-rollAnAce.js:
```diff
@@ -12,21 +12,21 @@ Full description at: https://github.com/HackYourFuture/Homework/blob/main/3-Usin
-  rollDieUntil('ACE')
-    .then((results) => console.log('Resolved!', results))
-    .catch((error) => console.log('Rejected!', error.message));
+async function main() {
+  try {
+    const results = await rollDieUntil('ACE');
+    console.log('Resolved!', results);
+  } catch (error) {
+    console.log('Rejected!', error.message);
+  }
 }
```
Perfect! 💥

nori0aw commented on 3-UsingAPIs/Week2/homework/ex4-diceRace.js:
```diff
@@ -15,14 +15,18 @@ const rollDie = require('../../helpers/pokerDiceRoller');
 function rollDice() {
   const dice = [1, 2, 3, 4, 5];
-  // TODO complete this function; use Promise.race() and rollDie()
+  const map1 = dice.map((x) => rollDie(x));
```
It is a good practice to use meaningful names when creating variables (map1& x)

✏️need to modify✏️

nori0aw commented on 3-UsingAPIs/Week2/homework/ex4-diceRace.js:
```diff
@@ -15,14 +15,18 @@ const rollDie = require('../../helpers/pokerDiceRoller');
+    console.log('Rejected!', error.message);
+  }
 }
 
```
You missed adding your explanation for the below:
“You may observe that some dice continue rolling for some undetermined time after the promise returned by Promise.race() resolves. Do you know why? Add your answer as a comment to the bottom of the file.”

✏️need to modify✏️

Sally-Samaan replied:
Because of the dice.map(), one for each die roll, and then passes this array to Promise.race(). This means that each promise is resolved independently and asynchronously, and there is no guarantee that all promises will complete at the same time. Therefore, some dice may continue rolling after the fastest die has already resolved.

nori0aw commented on 3-UsingAPIs/Week2/homework/ex5-vscDebug.js:
```diff
@@ -23,7 +25,7 @@ function renderLaureates(laureates) {
+    const { laureates } = await getData(
       'http://api.nobelprize.org/2.0/laureates?birthCountry=Netherlands&format=json&csvLang=en'
     );
     renderLaureates(laureates);
```
Works as expected! 💥

nori0aw commented on 3-UsingAPIs/Week2/homework/ex6-browserDebug/index.js:
```diff
@@ -43,7 +53,7 @@ function renderLaureates(laureates) {
+    const { laureates } = await getData(
       'https://api.nobelprize.org/2.0/laureates?birthCountry=Netherlands&format=json&csvLang=en'
     );
     renderLaureates(laureates);
```
Nice woke!💥

nori0aw reviewed (commented):
Well done Sally for doing most of the exercises correctly. 👏
Exercise 1: works just fine, I added a small comment to be modified. 
Exercise 2: needs some additional work. Please check my comments.
Exercise 3, 5, 6: all are perfect.. 💯 
Exercise 4: works as expected, you missed to answer the question. And I added a small comment.

Please check my comments. do not hesitate to reach out to me if you find anything not clear.

Sally-Samaan commented:
> Well done Sally for doing most of the exercises correctly. 👏 Exercise 1: works just fine, I added a small comment to be modified. Exercise 2: needs some additional work. Please check my comments. Exercise 3, 5, 6: all are perfect.. 💯 Exercise 4: works as expected, you missed to answer the question. And I added a small comment.
> 
> Please check my comments. do not hesitate to reach out to me if you find anything not clear.

Thank you so much for you comments @nori0aw , I will check and fix the codes after Wednesday, because now i am finishing the individual task for this week :)
